src/analysis/normed_space/conformal_linear_map.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import geometry.euclidean.basic
+import analysis.normed_space.inner_product
+
+/-!
+# Conformal Linear Maps
+
+A continuous linear map between `R`-normed spaces `X` and `Y` `is_conformal_map` if it is
+a nonzero multiple of a linear isometry.
+
+## Main results
+
+* `is_conformal_map`: the main definition
+* The conformality of the composition of two conformal linear maps, the identity map
+  and multiplications by nonzero constants as continuous linear maps
+* `is_conformal_map_iff`: an equivalent definition of the conformality
+* `is_conformal_map.preserves_angle`: if a continuous linear map is conformal, then it
+                                      preserves all angles in the normed space
+
+## Tags
+
+conformal
+
+## Warning
+
+The definition of conformality in this file does NOT require the maps to be orientation-preserving.
+-/
+
+noncomputable theory
+
+open linear_isometry continuous_linear_map
+open_locale real_inner_product_space
+
+/-- A continuous linear map `f'` is said to be conformal if it's
+    a nonzero multiple of a linear isometry. -/
+def is_conformal_map {R : Type*} {X Y : Type*} [nondiscrete_normed_field R]
+  [normed_group X] [normed_group Y] [normed_space R X] [normed_space R Y]
+  (f' : X →L[R] Y) :=
+∃ (c : R) (hc : c ≠ 0) (li : X →ₗᵢ[R] Y), (f' : X → Y) = c • li
+
+variables {E F : Type*} [inner_product_space ℝ E] [inner_product_space ℝ F]
+  {X Y Z : Type*} [normed_group X] [normed_group Y] [normed_group Z]
+  [normed_space ℝ X] [normed_space ℝ Y] [normed_space ℝ Z]
+  {R M N G : Type*} [nondiscrete_normed_field R]
+  [normed_group M] [normed_group N] [normed_group G]
+  [normed_space R M] [normed_space R N] [normed_space R G]
+
+lemma is_conformal_map_const_smul {c : R} (hc : c ≠ 0) : is_conformal_map (c • (id R M)) :=
+⟨c, hc, id, by ext; simp⟩
```

Comment:
Can you add `is_conformal_map_id` also, and use it in the proof of `conformal_at_id`.